### PR TITLE
refactor EmailValidator to use precompiled regex and simplify validation logic

### PR DESCRIPTION
**Pull request by fxrstor**
## Description
the previous implementation used ad-hoc split() and ad-hoc quote handling which made the logic hardto follow and errorprone. This change puts use to a precompiled pattern (faster, easier to read), enforces conservative but practical rules (no leading/trailing/consecutive dots in local part, label rules for domain, TLD sanity checks, punycode support), and sets practical length limits (trim + 6..254). The validator intentionally excludes extremely permissive RFC edge cases (quoted local parts, comments, etc.) that commonly break UIs and payment rails.

The validator becomes stricter only on exotic RFC constructs (quoted strings, nested comments). It accepts all commonly used addresses while removing flaky, UI-breaking cases.

## Changes
- Introduced SIMPLE_EMAIL_PATTERN (precompiled pattern) and replaced the previous multi-step string logic.
- Added explicit TLD checks (min length 2, letters-only or punycode xn--).
- Replaced instance invalidAddress with a constant INVALID_ADDRESS.
- Cleaned up input trimming and length constraints.

## Related issues
No related issues, just a general improvement.

## Summary by CodeRabbit

* **Refactor**
  * Tightened email validation: trimming, practical length limits, stricter local-part and TLD rules, and improved handling of international (punycode) TLDs.
* **Bug Fixes**
  * Reduced false positives/negatives for edge cases (extra whitespace, dots, hyphens, missing/short TLDs).
* **Tests**
  * Added comprehensive tests covering valid and invalid email scenarios to ensure consistent behavior.

### CODE REVIEW

**Comment by HenrikJannsen** [comment]
Would you mind to add a test class as well?

**Comment by fxrstor** [comment]
@HenrikJannsen Sorry, I totally forgot about this PR. I've added the test class. Can you review it?

**Comment by fxrstor** [comment]
@alvasw hey man, can you please review this?

File: desktop/src/main/java/bisq/desktop/util/validation/EmailValidator.java [hunk 1]
```diff
@@ -20,91 +20,89 @@
...
-
-        // check for odd number of double quotes before first and after last '@'
-        if ((subStrings[0].split("\"", -1).length % 2 == 1) || (subStrings[length].split("\"", -1).length % 2 == 1))
-            return false;
-        for (int k = 1; k < length; k++) {
-            if (subStrings[k].split("\"", -1).length % 2 == 0)
-                return false;
+        if (input == null) return INVALID_ADDRESS;
+
+        String email = input.trim();
```
Comment (wodoro) [inline desktop/src/main/java/bisq/desktop/util/validation/EmailValidator.java:70]: Validator trims the input, but caller stores the untrimmed version:
https://github.com/bisq-network/bisq/blob/76c10af839c8a11c0d61d76079b0739a56a62920/desktop/src/main/java/bisq/desktop/components/paymentmethods/WesternUnionForm.java#L160
 https://github.com/bisq-network/bisq/blob/76c10af839c8a11c0d61d76079b0739a56a62920/desktop/src/main/java/bisq/desktop/components/paymentmethods/WesternUnionForm.java#L193

So its better to not trim before validation here or fix all usages outside the validator
Validator must agree with what gets persisted.

File: desktop/src/main/java/bisq/desktop/util/validation/EmailValidator.java [hunk 2]
```diff
@@ -20,91 +20,89 @@
...
+     * - prevents leading/trailing/consecutive dots in the local part
+     * - ensures domain labels start/end with alnum and allows internal hyphens
+     */
+    private static final Pattern SIMPLE_EMAIL_PATTERN = Pattern.compile(
+            "^[A-Za-z0-9_%+\\-]+(?:\\.[A-Za-z0-9_%+\\-]+)*@" +          // local-part
+            "[A-Za-z0-9](?:[A-Za-z0-9\\-]{0,61}[A-Za-z0-9])?" +         // first domain label
+            "(?:\\.[A-Za-z0-9](?:[A-Za-z0-9\\-]{0,61}[A-Za-z0-9])?)+$"  // additional labels
+    );
+
+    private static final ValidationResult INVALID_ADDRESS = new ValidationResult(false, Res.get("validation.email.invalidAddress"));
```
Comment (wodoro) [inline desktop/src/main/java/bisq/desktop/util/validation/EmailValidator.java:57]: Resolve resource at runtime, not at creation time. (Don't call Res.get here)

File: desktop/src/main/java/bisq/desktop/util/validation/EmailValidator.java [hunk 3]
```diff
@@ -20,91 +20,89 @@
...
-
-    private final ValidationResult invalidAddress = new ValidationResult(false, Res.get("validation.email.invalidAddress"));
-
+    /**
+     * Precompiled pattern matching a conservative subset of email addresses.
+     * - prevents leading/trailing/consecutive dots in the local part
+     * - ensures domain labels start/end with alnum and allows internal hyphens
+     */
+    private static final Pattern SIMPLE_EMAIL_PATTERN = Pattern.compile(
+            "^[A-Za-z0-9_%+\\-]+(?:\\.[A-Za-z0-9_%+\\-]+)*@" +          // local-part
```
Comment (wodoro) [inline desktop/src/main/java/bisq/desktop/util/validation/EmailValidator.java:52]: I think it's better to follow RFC as much as we can. `'` is probably more common than `%`
overall better to only exclude what really breaks UI, if it there's any char in RFC that does, or even better, fix the UI to not break when showing them

(please mention what characters you have seen that breaks UI 100%)

**Review by wodoro — CHANGES_REQUESTED** [review]
Domain max length (253) not enforced. (only total 254 is enforced)
test: One giant validate() method, first assert failure hides the rest. break it into more tests or use `@ParameterizedTest`

**Comment by wodoro** [comment]
Hey @fxrstor 
any updates ? are you still interested in this PR? this sounds like a good change and we are willing to merge this if the mentioned issues are resolved

**Comment by fxrstor** [comment]
> Hey @fxrstor any updates ? are you still interested in this PR? this sounds like a good change and we are willing to merge this if the mentioned issues are resolved

Yes, I'm interested. It will take some time.

**Comment by alejandrogarcia83** [comment]
Hi @fxrstor,
I'm closing this PR for now. Please reopen it when you're ready.
